workflow-starfusion/src/main/java/ca/on/oicr/pde/workflows/STARfusionWorkflowClient.java

```diff
@@ -0,0 +1,189 @@
+ *
+ * See the SeqWare API for
+ * <a href="http://seqware.github.io/javadoc/stable/apidocs/net/sourceforge/seqware/pipeline/workflowV2/AbstractWorkflowDataModel.html#setupDirectory%28%29">AbstractWorkflowDataModel</a>
+ * for more information.
+ */
+public class STARfusionWorkflowClient extends OicrWorkflow {
+
+    //dir
+    private String dataDir, tmpDir;
+    private String outDir;
+
+    // Input Data
+    private String read1Fastq;
+    private String read2Fastq;
+    private String outputFilenamePrefix;
+  
+    // programs
+  //*  private String moduleFile;
+    
+    //Memory allocation
+    private Integer starfusionMem;
+ 
+
+    //path to bin
+    private String bin;
+
+    //ref Data
+    private String refGenome;
+
+
+    private boolean manualOutput;
+    private static final Logger logger = Logger.getLogger(STARfusionWorkflowClient.class.getName());
+    private String queue;
+    private Map<String, SqwFile> tempFiles;
+
+    // meta-types
+    private final static String TXT_METATYPE = "text/plain";
+    private final static String TAR_GZ_METATYPE = "application/tar-gzip";
+    private static final String FASTQ_GZIP_MIMETYPE = "chemical/seq-na-fastq-gzip";
+
+    private void init() {
+        try {
+            //dir
+            dataDir = "data";
+            tmpDir = getProperty("tmp_dir");
+            
+            //program
+           // moduleFile = getProperty("moduleLoadFile");
+
+            // input samples 
+            read1Fastq = getProperty("input_read1_fastq");
+            read2Fastq = getProperty("input_read2_fastq");
+
+            //Ext id
+            outputFilenamePrefix = getProperty("external_name");
+
+         
+            // ref fasta
+            refGenome = getProperty("ref_genome");
+           
+
+         
+            manualOutput = Boolean.parseBoolean(getProperty("manual_output"));
+            queue = getOptionalProperty("queue", "");
+
+            // starfusion
+                starfusionMem = Integer.parseInt(getProperty("starfusion_mem"));
+            
+            
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void setupDirectory() {
+        init();
+        this.addDirectory(dataDir);
+        this.addDirectory(tmpDir);
+        if (!dataDir.endsWith("/")) {
+            dataDir += "/";
+        }
+        if (!tmpDir.endsWith("/")) {
+            tmpDir += "/";
+        }
+    }
+
+    @Override
+    public Map<String, SqwFile> setupFiles() {
+        SqwFile file0 = this.createFile("read1");
+        file0.setSourcePath(read1Fastq);
+        file0.setType(FASTQ_GZIP_MIMETYPE);
+        file0.setIsInput(true);
+        SqwFile file1 = this.createFile("read2");
+        file1.setSourcePath(read2Fastq);
+        file1.setType(FASTQ_GZIP_MIMETYPE);
+        file1.setIsInput(true);
+        return this.getFiles();
+    }
+
+    @Override
+    public void buildWorkflow() {
+
+        /**
```

Comment:
This documentation section is for a different workflow, needs to be removed or edited